cmd/kubeadm/app/cmd/upgrade/plan.go

```diff
@@ -92,85 +156,63 @@ func runPlan(flags *planFlags, userVersion string) error {
 
 	// Compute which upgrade possibilities there are
 	klog.V(1).Infoln("[upgrade/plan] computing upgrade possibilities")
-	availUpgrades, err := upgrade.GetAvailableUpgrades(versionGetter, flags.allowExperimentalUpgrades, flags.allowRCUpgrades, etcdClient, cfg.DNS.Type, client)
+	availUpgrades, err := upgrade.GetAvailableUpgrades(versionGetter, flags.allowExperimentalUpgrades, flags.allowRCUpgrades, etcdClient, cfg.DNS.Type, client, printer)
```

Comment:
[2] It is a little bit weird that `GetAvailableUpgrades` require in input a printer, when few lines below we are calling `printAvailableUpgrades`

Reply:
Thank you for pointing this out. Will do.